zmq.hpp

```diff
@@ -138,11 +140,7 @@ class error_t : public std::exception
 {
   public:
     error_t() : errnum(zmq_errno()) {}
-#ifdef ZMQ_CPP11
-    virtual const char *what() const noexcept { return zmq_strerror(errnum); }
-#else
-    virtual const char *what() const throw() { return zmq_strerror(errnum); }
-#endif
+    virtual const char *what() const ZMQ_NOTHROW ZMQ_OVERRIDE { return zmq_strerror(errnum); }
```

Comment:
This changes the signature of what for pre-C++11, since ZMQ_NOTHROW is not defined as throw() but as nothing.

I am not sure if ZMQ_NOTHROW can be defined as throw() (I don't know at which other places the macro is used as of now), or if a specific macro should be defined for this case.

It is a good idea to remove the duplication here though.

Reply:
Actually, the exception specification is determined by the base class (i.e. the C++ standard), so in cannot be changed. The change breaks the build, see e.g. here: https://travis-ci.org/zeromq/cppzmq/jobs/505687621#L1401

Reply:
Fair enough. I'll try to replace the macro with `throw()` and see what happens.

Reply:
Fair enough.

Other usages are for non-virtual memfns. And most of them inline.

I'll try to replace the macro with throw() and see whether Travis complains.

Reply:
Ok, should be fine to add the nothrow declarations to these methods. Build is fine now. Thanks!